tests/ttnn/unit_tests/test_reshape.py

```diff
@@ -285,3 +285,45 @@ def test_reshape_tile_layout_only_change_shape(device):
     output = ttnn.to_torch(ttnn_output)
 
     assert_with_pcc(torch_result, output, 0.9999)
+
+
+# Reshape in Tile layout with shapes that are not divisible by 32
+@pytest.mark.parametrize(
+    "input_shape, output_shape",
+    [
+        ((1, 256, 16), (16, 256)),
+        ((1, 256, 1024), (1, 256, 16, 64)),
+        ((16, 16), (32, 8)),
+    ],
+)
+def test_reshape_tile_with_padding(input_shape, output_shape, device):
+    torch_input_tensor = torch.randn(input_shape, dtype=torch.bfloat16)
+    torch_result = torch_input_tensor.reshape(output_shape)
+
+    input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16, device=device)
+    ttnn_output = ttnn.reshape(input_tensor, output_shape)
+
+    output = ttnn.to_torch(ttnn_output)
+
+    assert_with_pcc(torch_result, output, 0.9999)
+
+
+# Since Inner dim is 1 of bfloat16, can't do on device, testing fallback on host
+@pytest.mark.parametrize(
+    "input_shape, output_shape",
+    [
+        ((1, 256, 1), (1, 256)),
+        ((1, 1024, 1), (1, 4, 256)),
+        ((1, 128, 1), (1, 128)),
+    ],
+)
+def test_reshape_host(input_shape, output_shape, device):
+    torch_input_tensor = torch.randn(input_shape, dtype=torch.bfloat16)
+    torch_result = torch_input_tensor.reshape(output_shape)
+
+    input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16, device=device)
```

Comment:
this test will fail if attempted in Row Major, right?

Reply:
it shouldn't as the Row major path is there.
but ya that's my bad on the test, I'll parameterize the input layout to cover both.

Reply:
Actually, I separated the two tests, actually one for tensors ending in shape 1 , and one for without.
The ones not ending in 1 I can do both TILE_LAYOUT and ROW_MAJOR_LAYOUT/

You are right this will fail but that's not due to reshape. it will fail in the creation of the torch tensor independent of reshape, hence this has to be in TILE_LAYOUT